CHANGELOG.md

```diff
@@ -1,5 +1,53 @@
+  * This will also allow you to attribute events to variations from other Optimizely projects in your account, even though those experiments don't appear in the same datafile.
+  * Note that for results segmentation in Optimizely results, the user attribute values from one event are automatically applied to all other events in the same session, as long as the events in question were actually received by our backend. This behavior was already in place and is not affected by the 3.0 release.
+* Support for all types of attribute values, not just strings.
+  * All values are passed through to notification listeners.
+  * Strings, booleans, and valid numbers are passed to the event dispatcher and can be used for Optimizely results segmentation. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³]. 
+  * Strings, booleans, and valid numbers are relevant for audience conditions.
+* Support for additional matchers in audience conditions:
+  * An `exists` matcher that passes if the user has a non-null value for the targeted user attribute and fails otherwise.
+  * A `substring` matcher that resolves if the user has a string value for the targeted attribute.
+  * `gt` (greater than) and `lt` (less than) matchers that resolve if the user has a valid number value for the targeted attribute. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³].
+  * The original (`exact`) matcher can now be used to target booleans and valid numbers, not just strings.
+* Support for A/B tests, feature tests, and feature rollouts whose audiences are combined using `"and"` and `"not"` operators, not just the `"or"` operator.
+* Datafile-version compatibility check: The SDK will remain uninitialized (i.e., will gracefully fail to activate experiments and features) if given a datafile version greater than 4.
+* Updated Pull Request template and commit message guidelines.
+* Only remove old variation assignments from the user profile service when there are >= 100 persisted assignments.
+* Support for empty bucketing IDs when evaluating feature rollouts.
+
+### Breaking Changes:
+* Previously, notification listeners were only given string-valued user attributes because only strings could be passed into various method calls. That is no longer the case. The `ActivateListener` and `TrackListener` interfaces now receive user attributes as `NSDictionary<NSString *, NSObject *> * _Nullable` instead of  `NSDictionary<NSString *,NSString *> * _Nonnull`.
+* The Localytics integration now relies on the user profile service to attribute events to the user's variations.
+* Drops support for the `getVariableBoolean`, `getVariableDouble`, `getVariableInteger`, and `getVariableString` APIs. Please migrate to the Feature Management APIs which are available since 2.x releases.
```

Comment:
My personal recommendation is
```suggestion
* Drops support for the `variableBoolean`, `variableDouble`, `variableInteger`, and `variableString` APIs. Please migrate to the Feature Management APIs which were introduced in the 2.x releases.
```
, but at the very least we should update the names of the APIs that we're removing support for. My fault - I initially forgot to adapt these when I copied the Android suggestion to Objective-C!